File: jee/java/src/main/java/fables/kotlin/jee/rest/RestApplication.java
package fables.kotlin.jee.rest;

import javax.ws.rs.ApplicationPath;
import javax.ws.rs.core.Application;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

/**
 * REST application configuration.
 *
 * @author Zeljko Trogrlic
 */
@ApplicationPath("api")
public class RestApplication
    extends Application {

  private final Set<Class<?>> classes = new HashSet<>(Arrays.asList(KittenRestService.class));

  @Override
  public Set<Class<?>> getClasses() {
    return this.classes;
  }
}

File: jee/converted/src/main/kotlin/fables/kotlin/jee/rest/RestApplication.kt
package fables.kotlin.jee.rest

import java.util.*
import javax.ws.rs.ApplicationPath
import javax.ws.rs.core.Application

/**
 * REST application configuration.
 * @author Zeljko Trogrlic
 */
@ApplicationPath("api")
class RestApplication : Application() {

    private val classes = HashSet<Class<*>>(Arrays.asList<Class<KittenRestService>>(KittenRestService::class.java))

    override fun getClasses(): Set<Class<*>> {
        return this.classes
    }
}
